lib/matplotlib/rcsetup.py

```diff
@@ -299,6 +301,12 @@ def validate_color(s):
 
     raise ValueError('%s does not look like a color arg%s' % (s, msg))
 
+def deprecate_axes_colorcycle(value):
+    warnings.warn("axes.color_cycle is deprecated.  Use axes.prop_cycle "
+                  "instead. Will be removed in 2.1.0")
+    return validate_colorlist(value)
```

Comment:
I agree with the first point, but on the second you misunderstand me, I didn't mention setting ``prop_cycle`` in ``validate_colorlist``, but in this ``deprecate_axes_colorcycle`` function.  As by definition this function will get removed in 2.1.0, it makes sense to me to that here.

I find it good practice to work around the deprecated stuff as early, aka as high up as possible in the codebase, rather then let it filter through to the deeper level code, as I see happens just above @tacaswell's comment "Should probably tag next version of cycler".